### ISSUE TITLE
Suspicious logic in WowUp.WPF.Utilities.FileUtilities.DeleteDirectory method

### ISSUE THREAD

**Issue by Phoenix-** [jliddev/WowUp#209]
First, you check the existence of target folder and delete it with force key(no questions here)
Then, you make 10 attempts every 100 msec to check the existence of the folder and fail if it still exists after 10 attempts.
Is this intended?
Probably, you want to make 10 attempts to delete folder and check its existence after everty attempt.
Checked at commit ec963b6fe5bd9447791eba909a6b3ea6d690d6c3 aka v1.18.4 tag

**Comment by jliddev** [comment]
Issues were found on slower file systems that if you dont wait for the exists check to fail then the code will fail to copy new things with the same path due to the underlying system not thinking the folder is gone yet.

**Comment by Phoenix-** [comment]
Okay, I got the point.
Thanks for the explanation.
I created tyis issue mainly because I constantly get 
`2020-10-23 18:32:42.537 +03:00 [ERR] Failed to copy addon directory C:/Games/World of Warcraft\_retail_\Interface\AddOns\ElvUI
System.Exception: Failed to delete directory
   at WowUp.WPF.Utilities.FileUtilities.DeleteDirectory(String filePath)
   at WowUp.WPF.Services.AddonService.<>c__DisplayClass49_0.<<InstallUnzippedDirectory>b__0>d.MoveNext()`
That is most likey because I use a symlink to a web drive instead of ordinary addons folder, but I doubt investigating such setups is worth your time.
